File: Linkedlist.java
class node{
    int data;
    node next;

    node(int data){
        this.data = data;
        this.next = null;
    }
}

class Linkedlist{
    node head =null;

    void insertatbeg(int data){
        node newnode = new node(data);
        newnode.next = head;
        head = newnode;
    }
    
    void insertAtPosition(int index,int data){
        if(head == null)
        {
            insertatbeg(data);
        }
        node temp = head;
        for(int i = 0; i < index - 1; i++)
        {
            temp = temp.next;
        }
        node newnode = new node(data);
        newnode.next = temp.next;
        temp.next = newnode;
    }

    void display(){
        node temp = head;
        while(temp != null)
        {
            System.out.print(temp.data+"->");
            temp = temp.next;
        }
        System.out.print("NUll");
    }

    public static void main(String [] args)
    {
        Linkedlist list = new Linkedlist();
        list.insertatbeg(10);
        list.insertatbeg(20);
        list.insertatbeg(30);
        list.insertAtPosition(2,25);
        list.display();
    }

}

File: linkedlist.py
class Node :
  def __init__ (self, data):
         self.data = data
         self.next = None
class linkedlist:
    def __init__(self):
        self.head = None


    def insert(self, data):
        new_node = Node(data)
        new_node.next = self.head
        self.head = new_node
        print("Node inserted successfully")

        
    def delete(self):
        if self.head is None:
            print("List is empty")
        else:
            self.head = self.head.next
            print("Node deleted successfully")


    def traverse(self):
        if self.head is None:
            print("List is empty")
        else:
            temp = self.head
            while temp is not None:
                print(temp.data, end=" -> ")
                temp = temp.next
            print("None")
l=[10,20,30]
l = linkedlist()

while True:
    print("\n--- LINKED LIST ---")
    print("1. Insert")
    print("2. Delete")
    print("3. Traverse")
    print("4. Exit")

    choice = int(input("Enter your choice: "))

    if choice == 1:
        data = int(input("Enter data: "))
        l.insert(data)

    elif choice == 2:
        l.delete()

    elif choice == 3:
        l.traverse()

    elif choice == 4:
        break

    else:
        print("Invalid choice")

